compiler/rustc_lint/src/non_fmt_panic.rs

```diff
@@ -52,13 +52,25 @@ impl<'tcx> LateLintPass<'tcx> for NonPanicFmt {
                 if Some(def_id) == cx.tcx.lang_items().begin_panic_fn()
                     || Some(def_id) == cx.tcx.lang_items().panic_fn()
                     || Some(def_id) == cx.tcx.lang_items().panic_str()
+                    || Some(def_id) == cx.tcx.lang_items().unreachable_display()
                 {
                     if let Some(id) = f.span.ctxt().outer_expn_data().macro_def_id {
+                        let diagnostic_item = cx.tcx.get_diagnostic_name(id);
                         if matches!(
```

Comment:
A series of `[else] if matches!()` on the same value can just be written as a regular `match`.